cloudformation/slack-router-cfn-template.js

```diff
@@ -0,0 +1,94 @@
+const cf = require('@mapbox/cloudfriend')
+
+const Resources = {
+  SlackRouterApi: {
+    Type: 'AWS::ApiGatewayV2::Api',
+    Properties: {
+      Name: 'slack-router-api',
+      ProtocolType: 'HTTP',
+    },
+  },
+  SlackToLambdaIntegration: {
+    Type: 'AWS::ApiGatewayV2::Integration',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+      Description: 'Integration for Slack Slash Commands',
+      IntegrationType: 'AWS_PROXY',
+      IntegrationUri: cf.join('', [
+        'arn:',
+        cf.partition,
+        ':apigateway:',
+        cf.region,
+        ':lambda:path/2015-03-31/functions/',
+        cf.getAtt('SlackRouterLambda', 'Arn'),
+        '/invocations',
+      ]),
+      PayloadFormatVersion: '2.0',
+    },
+  },
+  SlackToLambdaRoute: {
+    Type: 'AWS::ApiGatewayV2::Route',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+      AuthorizationType: 'NONE',
+      RouteKey: 'POST /slack-to-lambda',
+      Target: cf.join('/', [
+        'integrations',
+        cf.ref('SlackToLambdaIntegration'),
+      ]),
+    },
+  },
+  SlackToLambdaDeploy: {
+    Type: 'AWS::ApiGatewayV2::Deployment',
+    DependsOn: 'SlackToLambdaRoute',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+    },
+  },
+  SlackRouterApiProdStage: {
+    Type: 'AWS::ApiGatewayV2::Stage',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+      AutoDeploy: 'true',
+      DeploymentId: cf.ref('SlackToLambdaDeploy'),
+      StageName: 'prod',
+    },
+  },
+  SlackRouterLambda: {
+    Type: 'AWS::Lambda::Function',
+    Properties: {
+      FunctionName: 'slack-router-lambda',
+      Handler: 'slack-router-lambda.handler',
+      Role: cf.sub(
+        'arn:aws:iam::${AWS::AccountId}:role/service-role/test-api-lambda-sns-role-6ndj69lt'
```

Comment:
I think we have to separate out the SNS roles per stack, instead of hard-coding this value, to use CloudFormation in the best possible way. : ) Let's talk about this. cc/ @ramyaragupathy

Reply:
Hi @arunasank, what do you mean by separate out the SNS roles per stack? This cfn template does not create any SNS resources.

@ramyaragupathy just to confirm we are creating a new role per Lambda function which I will then use to replace the one that is currently in line 62? The role that's there (`test-api-lambda-sns-role-...`) is the role that you initially created when we were starting the project.

Reply:
I've created a separate role for the Lambda function.